### DISCUSSION TITLE
[Toggle Transcoding] Turn off transcoding

### DISCUSSION THREAD

**Discussion by faceleg** [immich-app/immich#4752] [Feature Request]
### Option to turn off/limit transcoding

My library has 40K+ videos, some are large. They are already in the format I want. This is an "external" library that I have managed for years, using IMMICH to enable the family to view it.

- I only use IMMICH in the iOS app or in Chrome.
- My library is about 1.6TB in total, and is stored on a mirrored 3TB pair of HDD (with cloud backup etc as well)

Judging from the way IMMICH has been causing disk usage increases over the 5 days it's been transcoding + the remaining # of files to transcode, I'm going to get to 2.6+TB disk usage, effectively doubling the space taken by my original already properly encoded video files.

I understand some will respond "disk space is cheap" and yes, it is cheaper today than it has ever been. This doesn't mean I want to go through the exiting process of purchasing new HDD and replacing/migrating to them in order to make space for transcodes I don't intend to view.

## Question:

Is there a way for one to turn off transcoding of video?

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
You can configure or turn off transcoding in the admin settings.

**Reply by faceleg** [reply]
Thanks! I'll have another look, obviously missed this.

**Reply by jmbaggs** [reply]
admin settings -> video transcode settings -> transcode policy.

**Reply by faceleg** [reply]
@jmbaggs is there a way for me to "correctly" delete transcodes that completed before I turned the function off?

**Reply by Jimmy-Z** [reply]
I'd like to know that too, I now have a ton of "offline paths" in "repair".

**Reply by danmed** [reply]
Could really do with know this too.....

**Reply by Esokrates** [reply]
@bo0tzz Same usage case as OP and same question: is there any way to delete already transcoded videos? They take up a lot of space and are not needed here because all my devices/browsers support the codecs I have.

**Comment by 518512** [comment]
Same usage case as OP and same question: is there any way to delete already transcoded videos? They take up a lot of space and are not needed here because all my devices/browsers support the codecs I have.

**Reply by jrasm91** [reply]
Not currently, no

**Reply by IoSonoAndreaZ** [reply]
But can be safely deleted manually?

**Reply by bo0tzz** [reply]
On recent versions the transcoding job will delete the files for any assets where the transcoding policy determines no transcode is needed.

**Comment by DeltaTango69** [comment]
I have turned off  transcoding in the settings

![image](https://github.com/immich-app/immich/assets/82575543/c91f1635-a063-4fdd-bd91-9195fcb08c25)

But under jobs are waiting many videos to transcode. I have paused them.

![image](https://github.com/immich-app/immich/assets/82575543/61881331-4f6f-46cc-9d1f-8421f389d7f2)

**Reply by bo0tzz** [reply]
The job will still run for all the assets, it just won't do anything.

**Reply by DeltaTango69** [reply]
When I push resume then all vidoes will be transcoded. No matter what the transcode policy says

**Reply by delicon** [reply]
After you disable the transcoding, click Clear on the job and than click All.
This will scan all transcoded videos and delete them.

**Reply by Snuupy** [reply]
It looks like after disabling transcoding, you have to click "all" and let it run, not clear

if you let it run, you will get:


```
immich_microservices     | [Nest] 7  - 03/02/2024, 12:02:31 PM     LOG [MediaService] Transcoded video exists for asset 1b2cd678-4940-450b-bb32-2f66ef62651f, but is no longer required. Deleting...
...
```

but if you clear it it will stop deleting the files

**Comment by 825i** [comment]
Can we please just have an option in the settings SPECIFICALLY that says **"Delete all transcoded videos"** with an "Are you sure?" confirmation.

Please please please!

Hunting through my entire library to try and identify which videos are transcoded or not is a total nightmare. This setting was ON by default when I installed Immich and it transcoded MANY videos without my knowledge...

I would also put the setting "Transcode Policy" at the **VERY TOP** of the settings pane for transcoding instead of over halfway down the bottom of the page (where it currently is). Transcoding is a massively resource intensive task which otherwise grinds your server to a near total halt. For those of us with thousands or even tens of thousands of videos, even if we ticked every single option to not transcode eg. x264, HEVC and so on. There can still be thousands of older videos *cough* quicktime *cough* that will be transcoded...

Please make the settings more user-friendly. The fact that this even forced-on by default is also something I argue against.

Please and thank you!

**Reply by 825i** [reply]
I will open a separate feature request for this as this ticket is closed.

**Comment by IoSonoAndreaZ** [comment]
i wonder if there is a way to know if i REALLY need transcoding. Is the app able to open any kind of format that modern phones create? Raw, heic, heivc and so on? What about chrome and firefox? Maybe I just need an on-the-fly transcode of a specific item only if I try to load it in an unsupported browser, this could also be an interesting option. Plex transcodes only if needed, for instance.

**Comment by 825i** [comment]
Opened feature request #10993

Please vote on it if you think it is a good idea. Thanks!
